src/web/gwc/src/main/java/org/geoserver/gwc/web/layer/UnconfiguredCachedLayersProvider.java

```diff
@@ -70,64 +157,34 @@ public int compare(TileLayer o1, TileLayer o2) {
      */
     @Override
     protected List<TileLayer> getItems() {
-        final GWC gwc = GWC.get();
-        final GWCConfig defaults = gwc.getConfig().saneConfig().clone();
-        final GridSetBroker gridsets = gwc.getGridSetBroker();
-        final Catalog catalog = getCatalog();
-
-        defaults.setCacheLayersByDefault(true);
-
-        List<String> unconfiguredLayerIds = getUnconfiguredLayers();
-
-        List<TileLayer> layers =
-                Lists.transform(
-                        unconfiguredLayerIds,
-                        new Function<String, TileLayer>() {
-                            @Override
-                            public TileLayer apply(String input) {
-                                GeoServerTileLayer geoServerTileLayer;
-
-                                LayerInfo layer = catalog.getLayer(input);
-                                if (layer != null) {
-                                    geoServerTileLayer =
-                                            new GeoServerTileLayer(layer, defaults, gridsets);
-                                } else {
-                                    LayerGroupInfo layerGroup = catalog.getLayerGroup(input);
-                                    geoServerTileLayer =
-                                            new GeoServerTileLayer(layerGroup, defaults, gridsets);
-                                }
-                                /*
-                                 * Set it to enabled regardless of the default settins, so it only shows up
-                                 * as disabled if the actual layer/groupinfo is disabled
-                                 */
-                                geoServerTileLayer.getInfo().setEnabled(true);
-                                return geoServerTileLayer;
-                            }
-                        });
-
-        return layers;
+        LOGGER.info("should not be called, fullSize() and getFilteredItems() overridden");
+        return unconfiguredLayers(Predicates.acceptAll())
+                .map(this::createUnconfiguredTileLayer)
+                .collect(Collectors.toList());
     }
 
-    private List<String> getUnconfiguredLayers() {
-        Catalog catalog = getCatalog();
-        List<String> layerIds = new LinkedList<String>();
-
-        GWC gwc = GWC.get();
+    private TileLayer createUnconfiguredTileLayer(PublishedInfo info) {
+        return new GeoServerTileLayer(info, defaults, GWC.get().getGridSetBroker());
+    }
 
-        List<LayerInfo> layers = catalog.getLayers();
-        for (LayerInfo l : layers) {
-            if (!gwc.hasTileLayer(l)) {
-                layerIds.add(l.getId());
-            }
-        }
+    private Stream<PublishedInfo> unconfiguredLayers(Filter filter) {
+        final Catalog catalog = getCatalog();
+        final CloseableIterator<LayerInfo> layers = catalog.list(LayerInfo.class, filter);
```

Comment:
Again just thinking out loud... there is maybe a potential to push down the offset/limit in the catalog, and chain back the groups iterator only when the first is exhausted, pushing down against offset and limit into it (the residual ones). Pretty complicated though, likely error prone and not worth it.